### PR TITLE
feat: Added types to `avm/res/cdn/profile` module

### LINKED ISSUE #3727: [AVM Module Issue]: Missing types in avm/res/cdn/profile
### Check for previous/existing GitHub issues

- [x] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/cdn/profile

### (Optional) Module Version

_No response_

### Description

Most types are missing for Azure Front Door in `avm/res/cdn/profile` module. Would be nice to have them added to improve developer experience.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by anderseide**
## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.cdn.profile](https://github.com/anderseide/avm-bicep-registry-modules/actions/workflows/avm.res.cdn.profile.yml) |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [X] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [X] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [X] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by gbeaud** [comment]
Thanks @anderseide for this impressive work! Please allow me some time to review as I'm currently under water but I will get back to you 👍

File: avm/res/cdn/profile/README.md
[the remark this answers is not in the document]
Comment (anderseide) [reply avm/res/cdn/profile/README.md:1]: Rebuilt using latest version of Bicep. Latest build pipeline: [avm.res.cdn.profile](https://github.com/anderseide/avm-bicep-registry-modules/actions/workflows/avm.res.cdn.profile.yml)

[the remark this answers is not in the document]
Comment (AlexanderSehr) [reply avm/res/cdn/profile/README.md:1]: Looking good @anderseide, thank you 💪 

I'll leave the rest of the review to @gbeaud 🚀

**Review by gbeaud — APPROVED** [review]
Good to go, thank you very much for the work! 🙏

**Pull request merged**